File: project/5.06_インスタンス変数とインスタンスメソッド/java/java_ex5_06_1/src/java_ex5_06_1/Ex5_06_1.java
package java_ex5_06_1;

class Sphere {
	double cx, cy, cz, radius;
	
	Sphere() {
		this( 0.0, 0.0, 0.0, 1.0 );		// コンストラクタを呼び出すことができる
	}
	
	Sphere( double radius ) {
		this( 0.0, 0.0, 0.0, radius );
	}
	
	Sphere( double cx, double cy, double cz, double radius ) {
		this.cx = cx;					// thisキーワードを用いてインスタンス変数にアクセス
		this.cy = cy;
		this.cz = cz;
		this.radius = radius;
	}
	
	// インスタンスメソッド
	void move( double cx, double cy, double cz ) {
		this.cx += cx;
		this.cy += cy;
		this.cz += cz;
	}
	
	void scale( double radius ) {
		this.radius = radius;
	}
	
}

public class Ex5_06_1 {

	public static void main(String[] args) {
		Sphere c1 = new Sphere();
		System.out.println( c1.cx + ", " + 
							c1.cy + ", " + 
							c1.cz + ", " +
							c1.radius );

		c1.move( 2, 2, 2 );
		System.out.println( c1.cx + ", " + 
							c1.cy + ", " + 
							c1.cz + ", " +
							c1.radius );
					
		c1.scale( 3.3 );
		System.out.println( c1.cx + ", " + 
							c1.cy + ", " + 
							c1.cz + ", " +
							c1.radius );
		
	}

}

File: project/5.06_インスタンス変数とインスタンスメソッド/py/py_ex5_06_1/src/ex5_06_1.py
'''
Created on 2023/05/03

@author: Keroichi.T
'''

class Sphere( object ):
    def __init__(self, cx = 0.0, cy = 0.0, cz = 0.0, r = 1.0):
        self.cx = cx
        self.cy = cy
        self.cz = cz
        self.r = r
    
    def move(self, cx, cy, cz):
        self.cx += cx
        self.cy += cy
        self.cz += cz
        
    def scale(self, radius):
        self.r = radius

def main():
    c1 = Sphere(1.2, 2.3, 3.4)
    print( f"{c1.cx}, {c1.cy}, {c1.cz}, {c1.r}" )
    
    c1.scale(5.5)
    print( f"{c1.cx}, {c1.cy}, {c1.cz}, {c1.r}" )

    c1.move(10,10,10)
    print( f"{c1.cx}, {c1.cy}, {c1.cz}, {c1.r}" )
    
    c2 = Sphere()
    print( f"{c2.cx}, {c2.cy}, {c2.cz}, {c2.r}" )    
    
if __name__ == '__main__':
    main()